go-controller/pkg/ovn/obj_retry.go

```diff
@@ -126,18 +189,1067 @@ func (r *retryObjs) getObjRetryEntry(key string) *retryObjEntry {
+	switch objectsToRetry.oType {
+	case factory.PodType:
+		var portInfo *lpInfo
+		pod := obj.(*kapi.Pod)
+		if cachedObj != nil {
+			portInfo = cachedObj.(*lpInfo)
+		}
+		oc.logicalPortCache.remove(util.GetLogicalPortName(pod.Namespace, pod.Name))
+		return oc.removePod(pod, portInfo)
+
+	case factory.PolicyType:
+		var cachedNP *networkPolicy
+		NP, ok := obj.(*knet.NetworkPolicy)
+		if !ok {
+			return fmt.Errorf("could not cast obj of type interface{} to *knet.NetworkPolicy")
+		}
+		if cachedObj != nil {
+			if cachedNP, ok = cachedObj.(*networkPolicy); !ok {
+				cachedNP = nil
+			}
+		}
+		if err := oc.deleteNetworkPolicy(NP, cachedNP); err != nil {
+			klog.Infof("Network Policy retry delete "+
+				"failed for %s/%s, will try again later: %v",
+				NP.Namespace, NP.Name, err)
+			return err
+		}
+
+	case factory.NodeType:
+		node, ok := obj.(*kapi.Node)
+		if !ok {
+			return fmt.Errorf("could not cast obj of type interface{} to *knet.Node")
+		}
+		if err := oc.deleteNodeEvent(node); err != nil {
+			klog.Infof("Node retry delete failed for %s, will try again later: %v",
+				node.Name, err)
+			return err
+		}
+
+	case factory.PeerServiceTypeVar:
+		service, ok := obj.(*kapi.Service)
+		if !ok {
+			return fmt.Errorf("could not cast peer service of type interface{} to *kapi.Service")
+		}
+		extraParameters, ok := objectsToRetry.extraParameters.(*NetworkPolicyExtraParameters)
+		if !ok {
+			return fmt.Errorf("could not cast extraParameters to " +
+				" *NetworkPolicyExtraParameters")
+		}
+		if err := oc.handlePeerServiceDelete(extraParameters.gp, service); err != nil {
+			klog.Infof("Peer service retry delete failed for %s, will try again later: %v",
+				service.Name, err)
+			return err
+		}
+
+	case factory.PeerPodSelectorTypeVar:
+		// TODO once you're done debugging, you can remove the casting here below
+		// and call handlePeerPodSelectorDelete with obj
+		pod, ok := obj.(*kapi.Pod)
+		if !ok {
+			return fmt.Errorf("could not cast peer pod of type interface{} to *kapi.Pod")
+		}
+		extraParameters, ok := objectsToRetry.extraParameters.(*NetworkPolicyExtraParameters)
+		if !ok {
+			return fmt.Errorf("could not cast extraParameters to " +
+				" *NetworkPolicyExtraParameters")
+		}
+		if err := oc.handlePeerPodSelectorDelete(extraParameters.gp, pod); err != nil {
+			klog.Infof("Peer pod retry delete failed for %s, will try again later: %v",
+				pod.Name, err)
+			return err
+		}
+
+	case factory.PeerNamespaceAndPodSelectorTypeVar:
+		extraParameters, ok := objectsToRetry.extraParameters.(*NetworkPolicyExtraParameters)
+		if !ok {
+			return fmt.Errorf("could not cast extraParameters to " +
+				" *NetworkPolicyExtraParameters")
+		}
+		// when the namespace labels no longer apply
+		// remove the namespaces pods from the address_set
+		var errs []error
+		namespace := obj.(*kapi.Namespace)
+		pods, _ := oc.watchFactory.GetPods(namespace.Name)
+
+		for _, pod := range pods {
```

Comment:
note to self, we dont remove the podHandler that we added previously (even in the old code). We should investigate if we should also be removing that from the handler list